### ISSUE TITLE
Long type names in deep directory structures cause AttachRuntimeDataSources to fail

### ISSUE THREAD

**Issue by phredaustin** [Thwaitesy/MSTestHacks#13]
When the combination of the test folder path and the type name exceeds the Windows 255 character limit, AttachRuntimeDataSources fails. This is because the source data gets stored in an xml file whose name is formed from type holding the data.

I've found that a partial solution is to use a hash of the data type name instead of the full type name as the name of the xml file.  This doesn't solve problems with length of the test folder path but if that's a problem you are seeing with your test project this solution at least won't make the problem worse.

If you are seeing this problem as we were due to very long fully qualified type names, this should help you.

The change I made was to replace line 97 in AttachRuntimeDataSources.cs with these lines:

`var saltedDataSourceName = "saltedDataSourceName" + dataSourceName + "saltedDataSourceName";
var dataSourceFilePath = Path.Combine(DATASOURCES_PATH, saltedDataSourceName.GetHashCode().ToString("X") + ".xml");`

(Salting the data source name may not be necessary here but I was concerned about data source type names that are really short possibly leading to hash collisions. I'm not really an expert in the area of hashing so this may not be necessary.)

I would make a pull request for this change but I don't know how to do this with Git so I'm hoping one of the project contributors will incorporate this change for me.

**Comment by Thwaitesy** [comment]
Hey @phredaustin - Thanks for locating this issue. I created a branch, updated the code, created a pull request and merged it. 

Ive used a guid instead of the salt + hashCode above. I dont think it makes much difference tho. 

Cheers,
Sam

**Comment by phredaustin** [comment]
That’s actually a much better solution than what I came up with.  Using guid’s completely eliminates the concern about hash collisions.


Thanks for the quick action,




Phred


Sent from Windows Mail





From: Sam Thwaites
Sent: ‎Monday‎, ‎August‎ ‎24‎, ‎2015 ‎10‎:‎21‎ ‎PM
To: Thwaitesy/MSTestHacks
Cc: Phred Menyhert





Hey @phredaustin - Thanks for locating this issue. I created a branch, updated the code, created a pull request and merged it. 

Ive used a guid instead of the salt + hashCode above. I dont think it makes much difference tho. 

Cheers,
Sam
